### PR TITLE
Updated compression to LZ4 

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow up of #54957, updated compression to LZ4 as its better in performance

### CODE REVIEW

File: src/Common/ZooKeeper/ZooKeeperImpl.cpp [hunk 1]
```diff
@@ -491,7 +491,7 @@ void ZooKeeper::connect(
                 if (use_compression)
                 {
                     compressed_in.emplace(*in);
-                    compressed_out.emplace(*out, CompressionCodecFactory::instance().get("ZSTD", {}));
+                    compressed_out.emplace(*out, CompressionCodecFactory::instance().get("LZ4", {}));
```
Comment (antonio2368) [inline src/Common/ZooKeeper/ZooKeeperImpl.cpp:494]: actually what do you think about making it a config also like `use_compression` and have `LZ4` be default?

Comment (SmitaRKulkarni) [reply src/Common/ZooKeeper/ZooKeeperImpl.cpp:494]: so we also exchange that in handshake ?

Comment (antonio2368) [reply src/Common/ZooKeeper/ZooKeeperImpl.cpp:494]: ah yes that part complicates it, let's have just LZ4 for now. 😓

**Review by antonio2368 — APPROVED** [review]

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 9a9efdcf7d39dfc585daa47a30fa384345d3a6d0 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56497/9a9efdcf7d39dfc585daa47a30fa384345d3a6d0/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6809661582">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56497/9a9efdcf7d39dfc585daa47a30fa384345d3a6d0/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**